src/form/remaining-chars-count/remaining-chars-count.directive.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * @ngdoc directive
+ * @name patternfly.form.directive:pfRemainingCharsCount
+ *
+ * @description
+ *   Directive for showing a characters remaining count and triggering warning and error
+ *   behavior when passing specified thresholds.
+ *
+ * @param {string} ng-model The scope model variable which contains the intial text for the text field.  Required, but</br>
+ * can be an emptly string ("").
+ * @param {string} count-fld The id of the field to display the 'characters-remaining' count.
+ * @param {string} chars-max-limit Number representing the maximum number of characters to allow before dispatching a<br/>
+ * 'overCharsMaxLimit' event.   When the number of characters falls below <code>chars-max-limit</code>, a 'underCharsMaxLimit'<br/>
+ * event is dispatched.
+ * @param {string} chars-warn-remaining Number of remaining characters to warn upon.  The 'chars-warn-remaining-pf'<br/>
+ * class will be applied to the <code>count-fld</code> when the remaining characters is less than the<br/>
+ * <code>chars-warn-remaining</code> threshold.  When/if the number of remaining characters becomes greater than the<br/>
+ * <code>chars-warn-remaining</code> threshold, the 'chars-warn-remaining-pf' class is removed from the <code>count-fld</code> field.
+ * @param {boolean=} block-input-at-max-limit If true, no more characters can be entered into the text field when the<br/>
+ * <code>chars-max-limit</code> has been reached.  If false (the default), characters may be entered into the text field after the<br/>
+ * max. limit has been reached, but these additional characters will trigger the 'overCharsMaxLimit' event to be<br/>
+ * dispatched.  When <code>blockInputAtMaxLimit</code> is <strong>true</strong>, a right-click 'paste' will only paste<br/>
+ * characters until the maximum character limit is reached.
+ *
+ * @example
+ <example module="patternfly.example">
+   <file name="index.html">
+     <div ng-controller="DemoCtrl" style="display:inline-block; width: 100%;">
+
+     <style>
+       textarea {
+         resize: none;
+       }
+     </style>
+
+     <div class="container">
+       <strong>Max Limit: 20, Warn When 5 Or less Remaining, Disable Button After Max Limit</strong>
+       <div class="row">
```

Comment:
No, for the title. Just a nit. "Max limit: 20. Warning when 5 or less remaining. Disable button after max limit reached."